arelle/PackageManager.py

```diff
@@ -253,7 +253,7 @@ def parsePackage(cntlr, filesource, metadataFile, fileBase, errors=[]):
                     prefixLength = len(mapFrom)
                     if prefixLength > longestPrefix:
                         _remappedUrl = remappedUrl[prefixLength:]
-                        if not (_remappedUrl[0] in (os.sep, '/') or mapTo[-1] in (os.sep, '/')):
+                        if len(_remappedUrl) > 0 and (not (_remappedUrl[0] in (os.sep, '/') or mapTo[-1] in (os.sep, '/'))):
```

Comment:
It seems easier to read with the `not` were distributed:
```
if len(_remappedUrl) > 0 and not _remappedUrl[0] in (os.sep, '/') and not mapTo[-1] in (os.sep, '/'):
```
Does the same issue also exist for empty `mapTo`?  Perhaps `startswith(tuple)`/`endswith(tuple)` is even better:
```
if len(_remappedUrl) > 0 and not _remappedUrl.startswith((os.sep, '/')) and not mapTo.endswith((os.sep, '/')):
```
Well, I'm somewhat surprised that `os.sep` is involved when it seems like URL remappings should always use `/`.

Reply:
Agreed on all points, and have implemented your second suggestion.

I'm not even sure what this code is doing.  It seems to do some fixing up of remappings that is not justified by the [spec](https://www.oasis-open.org/committees/download.php/14809/xml-catalogs.html#s.rewriteuri):

> Rewriting removes the matching prefix from the supplied URI reference and replaces it with the value of the value of the rewritePrefix attribute.

I don't  think there's anything that requires a remapping to split on a path segment boundary, and I can't see anything that suggests that `/` should be inserted.

Reply:
> I don't think there's anything that requires a remapping to split on a path segment boundary, and I can't see anything that suggests that / should be inserted.

I've raised a bug for the [Taxonomy Package spec](https://gitlab.xbrl.org/base-spec/taxonomy-package/-/issues/14) for this (XII gitlab, access restricted).